apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/network/http/ApolloHttpNetworkTransport.kt

```diff
@@ -8,5 +8,12 @@ import com.apollographql.apollo.network.NetworkTransport
 expect class ApolloHttpNetworkTransport constructor(
     serverUrl: String,
     headers: Map<String, String>,
-    httpMethod: HttpMethod = HttpMethod.Post
+    httpMethod: HttpMethod = HttpMethod.Post,
+    /**
+     * The timeout interval to use when waiting for additional data.
+     *
+     * - on iOS, it is used to set NSMutableURLRequest.timeoutInterval
+     * - on Android, it is used to set both readTimeout and connectTimeout
+     */
+    timeoutMillis: Long = 30_000
```

Comment:
should we have 2 timeouts for connection and reading, looks like NSURLsession allows to set both:
```
https://developer.apple.com/documentation/foundation/nsmutableurlrequest/1414063-timeoutinterval?language=objc
https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1408259-timeoutintervalforrequest?language=objc
```

Reply:
👍 I missed `timeoutIntervalForRequest`, will add that

Reply:
Mmm, `timeoutIntervale` is on `NSMutableUrlRequest` but `timeoutIntervalForRequest` is for `NSUrlSession`. I'm not sure we can have both at the same time?

Reply:
Looks like we can! I just pushed a few changes

Reply:
i'm not 100% sure the relation between `timeoutInterval` and `timeoutIntervalForRequest`. It might make sense to map everything to the same timeout on iOS. But on the other hand, having both is more flexible in terms of API so we can certainly keep it as it is, even if there's a small chance the concept do not map 1:1